lib/arn.go

```diff
@@ -11,7 +12,8 @@ import (
 //   arn:partition:service:region:account-id:resourcetype:resource
 
 var (
-	ErrNotAnArn = errors.New("Not an ARN")
+	ErrNotAnArn      = errors.New("Not an ARN")
+	RolePrefixRegexp = regexp.MustCompile(`^role/\w`)
```

Comment:
As this variable shouldn't be part of the `vaulted` library interface, we should change this to be lowercase. And since this is scoped to the "resource" component of the ARN, I think we should name it accordingly (right now it sounds like, to me anyway, we're trying to match against the ARN as a whole).

Additionally, this `regexp` is too restrictive (`@Potter,Harry` is a valid role name, for example). Perhaps we should just check for the existence of at least one additional character? (see **RoleName** in the [CreateRole API](https://docs.aws.amazon.com/IAM/latest/APIReference/API_CreateRole.html) documentation)